### PR TITLE
Possibility of disabling audit log functionality per landscape

### PR DESCRIPTION
**Pull request by mvshao**
**Description**

Changes proposed in this pull request:

- Audit Log can be configured as not strict requirement for specific landscapes

**Related issue(s)**
#286

### CODE REVIEW

**Comment by Disper** [comment]
/hold
should wait for #365

File: internal/controller/runtime/fsm/runtime_fsm_configure_auditlog.go [hunk 1]
```diff
@@ -24,19 +26,45 @@ func sFnConfigureAuditLog(ctx context.Context, m *fsm, s *systemState) (stateFn,
...
+		} else {
+			if m.RCCfg.AuditLogMandatory {
+				m.log.Error(err, "Failed to configure Audit Log")
+				s.instance.UpdateStatePending(
+					imv1.ConditionTypeAuditLogConfigured,
+					imv1.ConditionReasonAuditLogError,
+					"False",
+					errorMessage)
+			} else {
+				m.log.Info(errorMessage, "Failed to configure Audit Log, but is not mandatory to be configured")
```
Comment (Disper) [inline internal/controller/runtime/fsm/runtime_fsm_configure_auditlog.go:56]: During verification I've encountered a panic:
```
2024-09-09T15:29:14+02:00       INFO    Observed a panic in reconciler: odd number of arguments passed as key-value pairs for logging   {"controller": "runtime", "controllerGroup": "infrastructuremanager.kyma-project.io", "controllerKind": "Runtime", "Runtime": {"name":"kim-md-al6","namespace":"kcp-system"}, "namespace": "kcp-system", "name": "kim-md-al6", "reconcileID": "1aa26a9a-cc11-4cd7-bdf7-b6b022203af0"}
panic: odd number of arguments passed as key-value pairs for logging [recovered]
        panic: odd number of arguments passed as key-value pairs for logging

```
Quoting logger api docs:

> The key/value pairs must alternate string keys and arbitrary// values.

I wonder if just `m.log.Info(errorMessage)` wouldn't be enough here?

Comment (mvshao) [reply internal/controller/runtime/fsm/runtime_fsm_configure_auditlog.go:56]: I need to check code. In my opinion this additional info need to be present to clearly describe that Audit Log was set as not mandatory.

Comment (Disper) [reply internal/controller/runtime/fsm/runtime_fsm_configure_auditlog.go:56]: It's possible to add either one parameter (errorMessage) or 3/5/7 etc with additional ones for key name and the value . (always uneven). 

When I executed this part I noticed the `Error during enabling Audit Logs on shoot: kim-md-al8: missing mapping for selected region in provider config` was the errorMessage and passed as a string was`Failed to configure Audit Log, missing region mapping for this shoot, but is not mandatory to be configured` . which indeed contains an additional information that's mandatory. 

So how about such invocation: 
`m.log.Info(errorMessage, "AuditLogMandatory", m.RCCfg.AuditLogMandatory)`?

The log would look like this 
`2024-09-10T07:31:51+02:00       INFO    reqID 1 Error during enabling Audit Logs on shoot: kim-md-al8: missing mapping for selected region in provider config   {"AuditLogMandatory": false}`

Comment (Disper) [reply internal/controller/runtime/fsm/runtime_fsm_configure_auditlog.go:56]: Logging additional information on region and provider type could make it easier to quickly fix such issues. `m.log.Info(errorMessage, "AuditLogMandatory", m.RCCfg.AuditLogMandatory, "providerType", s.shoot.Spec.Provider.Type, "region", s.shoot.Spec.Region) ` could make the logs easier

**Review by Disper — APPROVED** [review]

**Pull request merged**